torch/fx/experimental/symbolic_shapes.py

```diff
@@ -274,6 +279,20 @@ def parallel_and(*args):
         return False
     return all(args)
 
+def sym_eq(x, y):
+    """
+    Like ==, but when run on list/tuple, it will recursively test equality
+    and use sym_and to join the results together, without guarding.
+    """
+    if (isinstance(x, tuple) and isinstance(y, tuple)) or (isinstance(x, list) and isinstance(y, list)):
```

Comment:
Can it be that one is list and other is tuple; and do we want to allow that, too?

Reply:
I'm matching `==` semantics, which says that list and tuple are never equal